src/operator/subgraph/subgraph_op.cc

```diff
@@ -154,10 +154,27 @@ bool SubgraphOpShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(idx_g.input_nodes().size(), in_shapes->size());
   CHECK_EQ(idx_g.outputs().size(), out_shapes->size());
   // TODO: make sure shape inputs matches the order from in_shapes
-  nnvm::ShapeVector shape_inputs = *in_shapes;
-  imperative::CheckAndInferShape(&g, std::move(shape_inputs), true);
-  const nnvm::ShapeVector& shapes = g.GetAttr<nnvm::ShapeVector>("shape");
-  const std::vector<uint32_t>& input_nids = idx_g.input_nodes();
+
+  // Put the input and output shapes to the shape vector.
+  nnvm::ShapeVector shapes(idx_g.num_nodes());
```

Comment:
You are populating known shapes with entry ids. I think this one should be allocated with size equal to `idx_g.num_node_entries()` and attache attribute `shape` to the graph.